src/response_decoder.ts

```diff
@@ -264,12 +264,12 @@ function decodeMessageHeader(dataResponse: DataReader) {
     throw new Error(`invalid composite header %#02x: ${type}`)
   }
 
-  const nextType = dataResponse.readInt8()
-  decodeFormatCode(dataResponse, nextType)
+  // next, the composite type is encoded as an AMQP uint8
+  dataResponse.readUInt64()
 
   const formatCode = dataResponse.readUInt8()
   const headerLength = decodeFormatCode(dataResponse, formatCode)
-  if (!headerLength) throw new Error(`invalid formatCode %#02x: ${formatCode}`)
+  if (headerLength === undefined) throw new Error(`invalid formatCode %#02x: ${formatCode}`)
```

Comment:
again

Reply:
Same as above, if headerLength is 0 it throws the exception, but 0 can be a correct value